src/main/java/nitezh/ministock/activities/GlobalWidgetData.java

```diff
@@ -23,4 +40,187 @@
     public void setGlobalList(List<WidgetRow> list) {
         myStockList = list;
     }
+
+    // Variables in Graph Implementation
+    public void setURLString(String urlString) {
+
+        this.urlString = urlString;
+    }
+
+    public String getURLString() {
+
+        return urlString;
+    }
+
+
+
+    // Used in Time Scale of Graph Implementation
+    public static void setInterval(int intervalVal) {
+
+        interval = intervalVal;
+    }
+
+    public static int getInterval() {
+        return interval;
+    }
+
+
+    // Methods for Graph Implementation
+    public static JsonObject getJsonObjectRoot(String sURL) {
+        JsonObject rootObj;
+        try {
+            URL url = new URL(sURL);
+            HttpURLConnection request = (HttpURLConnection) url.openConnection();
+            request.connect();
+
+            //Convert to a JSON object to print data
+            JsonParser jp = new JsonParser();                           //use gson
+            JsonElement root = jp.parse(new InputStreamReader((InputStream)
+                    request.getContent()));
+
+            // Get a handle on the root
+            rootObj = root.getAsJsonObject();
+            return rootObj;
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public static List<String> getValues(String sURL, int interval) {
+        List<String> toReturn = null;
+        try {
+            toReturn = new JsonSnatcher(sURL, interval).execute(sURL).get();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return toReturn;
+    }
+
+    public static String constructImageUrl(List<String> list) {
+        String cht;             //Chart Type
+        String chd;             //Chart Data  (i.e. t:Data1,Data2,Data2,etc)
+        String chds;            //Text Format custom scaling
+        String chxt;            //Visible Axes
+        String chs;             //Chart Size
+        String chtt;            //Chart Title
+        cht = "cht=ls";                                             //Line Graph
+        chd = "chd=t%3A";                                           //Data of line graph. Must begin with t:
+        chds = "chds=a";                                            //Automatic text format scaling
+        chxt = "chxt=x%2Cy";                                         //Specify X and Y Axes
+        chs = "chs=700x690";                                        //chart size
+        chtt = "chtt=Test%20Graph";                                 //Name of Graph
+        String chdVars = "";
```

Comment:
`chtt = "chtt=Test%20Graph";                                 //Name of Graph`

this should be renamed to the name of the actual graph. the name "test graph" was chosen for test cases

Reply:
Changed